bundles/models/models_service.go

```diff
@@ -838,3 +889,37 @@ func (ms *Service) createUniqueModelName(tx *gorm.DB, name, owner string) (strin
 	}
 	return newName, nil
 }
+
+// isbasicModelListQuery returns a boolean that indicates if this a basic `GET /models` or `GET /models?page=N` query.
+// In this case, we can ideally use the memdory cache to reduce the
+// DB burden.
+// Note: the PerPage default value is 20.
+func isbasicModelListQuery(p *gz.PaginationRequest, owner *string,
+	order, search string, likedBy *users.User, ignoreMemcache bool) bool {
+	return !ignoreMemcache && owner == nil && order == "" && search == "" && likedBy == nil && p != nil && (!p.PageRequested || (p.PageRequested && p.PerPage == 20))
+}
+
+// getModelListCache attempts to get a query result from memcache.
+func getModelListCache(basicQuery bool, modelsCacheKey, paginationCacheKey string) (*fuel.Models, *gz.PaginationResult, bool) {
+	if basicQuery {
```

Comment:
Considering simplifying this block by using something like:

```suggestion
	if !basicQuery {
		return nil, nil, false
	}
```

That way you can early return if it's not a basic query, and reduce one nested if

Reply:
There are three layers of nested `if`s that are designed to fall through to the final return statement. Thanks for the suggestion, but I'll leave this as is otherwise I'd have to change three layers of logic.